pkg/git/git.go

```diff
@@ -31,6 +32,7 @@ import (
 const (
 	sshKnownHostsUserPath          = ".ssh/known_hosts"
 	sshMissingKnownHostsSSHCommand = "ssh -o StrictHostKeyChecking=accept-new"
+	sshURLFormat                   = `.+@([\w\d\.]+)(:[\d]+){0,1}/*(.*)`
```

Comment:
From https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols

> To clone a Git repository over SSH, you can specify an ssh:// URL like this:
> 
> ```
> $ git clone ssh://[user@]server/project.git
> ```
> 
> Or you can use the shorter scp-like syntax for the SSH protocol:
> 
> ```
> $ git clone [user@]server:project.git
> ```
> 
> In both cases above, if you don’t specify the optional username, Git assumes the user you’re currently logged in as.

I'm not sure this regex captures all combinations (e.g. `user@` is optional. `:` can precede port or `<project>.git` in the scp syntax. It also looks like it's possible to have a url without `/`).  It would be great to introduce some testing to make sure we're covering as many different variations of this url as we can.  So it might make sense to perform the url check in a separate function that can be exercised independently of the `.ssh` directory checking.

Reply:
https://www.debuggex.com/r/LU8VCTiHXLMN1m8X